File: Python/hashtable/q1.py
from typing import List


class Solution1:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashtable = dict()
        for index, num in enumerate(nums):
            if target - num in hashtable:
                return [hashtable[target - num], index]
            hashtable[nums[index]] = index
        return []

File: Cpp/src/hashtable/q1.cpp
#include <vector>
#include <unordered_map>

namespace q1 {
    class Solution1 {
    public:
        std::vector<int> twoSum(std::vector<int> &nums, int target) {
            std::unordered_map<int, int> hashtable;
            for (int i = 0; i < nums.size(); ++i) {
                auto it = hashtable.find(target - nums[i]);
                if (it != hashtable.end()) {
                    return {it->second, i};
                }
                hashtable[nums[i]] = i;
            }
            return {};
        }
    };
}